### ISSUE TITLE
Unable  to verify

### ISSUE THREAD

**Issue by Chi-bz** [8go/matrix-commander#171]
Hi, I've been having issues trying to verify. I installed matrix-command through pip in a Python env and was able to successfully send a test message. However, I've been unable to get it verified trying both the Elements web-app in two different installations of Firefox, as well as their Flatpak app. 

Upon clicking the “Interactively verify by emoji” button through the room-info panel, the window for verification disappears and shows no signs of working. I was able to use the exact same method to verify other clients to confirm it wasn't an issue regarding the Elements apps. So I'm assuming there's an issue on the matrix-command side, whether that be user error or otherwise, I'm not quite sure.

```
$ matrix-commander --verify emoji
matrix-commander is ready and waiting for the other party to initiate an emoji verification with us by selecting 'Verify by Emoji' in their Matrix client. Read --verify instructions in --manual carefully to assist you in how to do this quickly.
2024-03-19 02:10:18,725:     INFO: nio.client.base_client: New joined room room_string:matrix.org
2024-03-19 02:10:18,726:     INFO:        nio.rooms: Room room_string:matrix.org handling event of type PowerLevelsEvent
2024-03-19 02:10:18,726:     INFO:        nio.rooms: Room room_string:matrix.org handling event of type RedactedEvent
2024-03-19 02:10:18,726:     INFO:        nio.rooms: Room room_string:matrix.org handling event of type RedactedEvent
2024-03-19 02:10:18,726:     INFO:        nio.rooms: Room room_string:matrix.org handling event of type RedactedEvent
2024-03-19 02:10:18,726:     INFO:        nio.rooms: Room room_string:matrix.org handling event of type RedactedEvent
2024-03-19 02:10:18,726:     INFO:        nio.rooms: Room room_string:matrix.org handling event of type RoomCreateEvent
2024-03-19 02:10:18,726:     INFO: nio.client.base_client: New joined room room_string:matrix.org
2024-03-19 02:10:18,726:     INFO:        nio.rooms: Room room_string:matrix.org handling event of type RoomHistoryVisibilityEvent
2024-03-19 02:10:18,726:     INFO:        nio.rooms: Room room_string:matrix.org handling event of type RoomCreateEvent
2024-03-19 02:10:18,726:     INFO:        nio.rooms: Room room_string:matrix.org handling event of type RoomEncryptionEvent
2024-03-19 02:10:18,726:     INFO:        nio.rooms: Room room_string:matrix.org handling event of type RoomGuestAccessEvent
2024-03-19 02:10:18,726:     INFO:        nio.rooms: Room room_string:matrix.org handling event of type PowerLevelsEvent
2024-03-19 02:10:18,726:     INFO:        nio.rooms: Changing power level for user @acct:matrix.org from 0 to 100
2024-03-19 02:10:18,726:     INFO:        nio.rooms: Changing power level for user @bot_acct:matrix.org from 0 to 100
2024-03-19 02:10:18,726:     INFO:        nio.rooms: Room room_string:matrix.org handling event of type RoomJoinRulesEvent
2024-03-19 02:10:18,726:     INFO: nio.client.base_client: New joined room room_string:matrix.org
2024-03-19 02:10:18,726:     INFO:        nio.rooms: Room room_string:matrix.org handling event of type RoomNameEvent
2024-03-19 02:10:18,726:     INFO:        nio.rooms: Room room_string:matrix.org handling event of type RoomSpaceParentEvent
2024-03-19 02:10:18,726:     INFO:        nio.rooms: Room room_string:matrix.org handling event of type RedactedEvent
2024-03-19 02:10:18,726:     INFO:        nio.rooms: Room room_string:matrix.org handling event of type RoomJoinRulesEvent
2024-03-19 02:10:18,726:     INFO:        nio.rooms: Room room_string:matrix.org handling event of type RoomEncryptionEvent
2024-03-19 02:10:18,726:     INFO:        nio.rooms: Room room_string:matrix.org handling event of type RoomHistoryVisibilityEvent
2024-03-19 02:10:18,726:     INFO:        nio.rooms: Room room_string:matrix.org handling event of type RoomGuestAccessEvent
2024-03-19 02:10:18,726:     INFO:        nio.rooms: Room room_string:matrix.org handling event of type RoomCreateEvent
2024-03-19 02:10:18,726:     INFO:        nio.rooms: Room room_string:matrix.org handling event of type PowerLevelsEvent
2024-03-19 02:10:18,726:     INFO:        nio.rooms: Changing power level for user @acct:matrix.org from 0 to 100

**Comment by 0x766C70** [comment]
got this message when initiating verification from browser:

ERROR: matrix-commander: E110: Verification has been cancelled by @user:matrix.localinstance.co, for reason "The user cancelled the verification.".

**Comment by spacefrogg** [comment]
I tried to use `--verify` together with `--login` and used the steps according to the `--manual`. This got me here:
```
2024-04-12 11:38:26,500:    ERROR: matrix-commander: E108: accept_key_verification failed with error 'ToDeviceError: M_UNKNOWN Internal server error - retry after ToDeviceMessage(type='m.key.verification.accept', recipient='<redacted>', recipient_device='<redacted>', content={'transaction_id': 'jBGTvgYL6RlbH7ejcWTXGZfD2UHEWVoB', 'key_agreement_protocol': 'curve25519-hkdf-sha256', 'hash': 'sha256', 'message_authentication_code': 'hkdf-hmac-sha256', 'short_authentication_string': ['emoji', 'decimal'], 'commitment': '<redacted>'})ms'.
2024-04-12 11:38:26,522:  WARNING:    nio.responses: Error validating response: Additional properties are not allowed ('errcode', 'error' were unexpected)
2024-04-12 11:38:26,522:    ERROR: matrix-commander: E109: to_device failed with error 'ToDeviceError: M_UNKNOWN Internal server error - retry after ToDeviceMessage(type='m.key.verification.key', recipient='<redacted>', recipient_device='<redacted>', content={'transaction_id': 'jBGTvgYL6RlbH7ejcWTXGZfD2UHEWVoB', 'key': '<redacted>'})ms'.
2024-04-12 11:39:26,663:    ERROR: matrix-commander: E110: Verification has been cancelled by <redacted> for reason "Timed out".
2024-04-12 11:39:26,757:  WARNING:    nio.responses: Error validating response: Additional properties are not allowed ('errcode', 'error' were unexpected)
```

The verification on the browser window keeps spinning endlessly (waiting for an emoji response from matrix-commander). It seems that matrix-commander is not compatible with the latest protocol and chokes on newly added fields `errcode` and `error`.

**Comment by garryavx** [comment]
Same issue, tried in element web on firefox and element desktop on win11. I initiate the verification request from the element client in the sessions section of settings, then run
matrix-commander -c /home/nagios/.config/matrix-commander/store/credentials.json -s /home/nagios/.config/matrix-commander/ --verify

I get
" ERROR: matrix-commander: E110: Verification has been cancelled by @nagiosrelay:matrix.org for reason "The user cancelled the verification."."
then it just hangs

**Comment by thehecki** [comment]
same problem for me, no reaction after start emoji verify

**Comment by TorstenC** [comment]
Similar problem:

    nio.crypto.log: Received key verification event with an unknown transaction id from @xxxx:matrix.xxxx.de

**Comment by oddfellow** [comment]
I used the Element Web Client from a different IP address than the one from matrix-commander and then emoji verification worked as described in `matrix-commander --manual`

Trying to verify from the same IP address resulted in the problems described above.

I could reproduce this two times in a row.

**Comment by bpbradley** [comment]
> I used the Element Web Client from a different IP address than the one from matrix-commander and then emoji verification worked as described in `matrix-commander --manual`
> 
> Trying to verify from the same IP address resulted in the problems described above.
> 
> I could reproduce this two times in a row.

Doesn't work for me. Having the same issues as described by others.

**Comment by tionis** [comment]
Same here

**Comment by SuperdukeGates** [comment]
> I used the Element Web Client from a different IP address than the one from matrix-commander and then emoji verification worked as described in `matrix-commander --manual`
> 
> Trying to verify from the same IP address resulted in the problems described above.
> 
> I could reproduce this two times in a row.

No help. :(

**Comment by HiFiJ** [comment]
Hi, I have been trying hard to get a successfull verififcation via emoji verification initiated by the Element Web App, but can not get any reaction on the matrx-commander side. Is there anything special regarding this combination or other clue on how to verify the matrix-commander device?
Thanks a lot, Joost

I tried on Manjaro and on up-to-date IPFire (via pip):

`   ~/Software/Python/matric-commander  matrix-commander --verify emoji                                                                                                                                                                                                                     1 ✘ 
matrix-commander is ready and waiting for the other party to initiate an emoji verification with us by selecting 'Verify by Emoji' in their Matrix client. Read --verify instructions in --manual carefully to assist you in how to do this quickly.
2024-05-26 19:55:25,210:     INFO: nio.client.base_client: New joined room !evcbAiNuuyEVFyAQjX:matrix.org
2024-05-26 19:55:25,211:     INFO:        nio.rooms: Room !evcbAiNuuyEVFyAQjX:matrix.org handling event of type RoomHistoryVisibilityEvent
2024-05-26 19:55:25,211:     INFO:        nio.rooms: Room !evcbAiNuuyEVFyAQjX:matrix.org handling event of type RoomJoinRulesEvent
2024-05-26 19:55:25,211:     INFO:        nio.rooms: Room !evcbAiNuuyEVFyAQjX:matrix.org handling event of type RoomEncryptionEvent
2024-05-26 19:55:25,211:     INFO:        nio.rooms: Room !evcbAiNuuyEVFyAQjX:matrix.org handling event of type RoomNameEvent
2024-05-26 19:55:25,212:     INFO:        nio.rooms: Room !evcbAiNuuyEVFyAQjX:matrix.org handling event of type PowerLevelsEvent
2024-05-26 19:55:25,212:     INFO:        nio.rooms: Changing power level for user @j.c.w.:matrix.org from 0 to 100
2024-05-26 19:55:25,212:     INFO:        nio.rooms: Room !evcbAiNuuyEVFyAQjX:matrix.org handling event of type RoomCreateEvent
2024-05-26 19:55:25,212:     INFO:        nio.rooms: Room !evcbAiNuuyEVFyAQjX:matrix.org handling event of type RoomGuestAccessEvent
2024-05-26 19:55:25,212:     INFO: nio.client.base_client: New joined room !dWitBphSDydauUWqIi:matrix.org
2024-05-26 19:55:25,212:     INFO:        nio.rooms: Room !dWitBphSDydauUWqIi:matrix.org handling event of type RoomCreateEvent
2024-05-26 19:55:25,212:     INFO:        nio.rooms: Room !dWitBphSDydauUWqIi:matrix.org handling event of type RoomJoinRulesEvent
2024-05-26 19:55:25,212:     INFO:        nio.rooms: Room !dWitBphSDydauUWqIi:matrix.org handling event of type PowerLevelsEvent
2024-05-26 19:55:25,213:     INFO:        nio.rooms: Room !dWitBphSDydauUWqIi:matrix.org handling event of type RoomEncryptionEvent
2024-05-26 19:55:25,213:     INFO:        nio.rooms: Room !dWitBphSDydauUWqIi:matrix.org handling event of type RoomGuestAccessEvent
2024-05-26 19:55:25,213:     INFO:        nio.rooms: Room !dWitBphSDydauUWqIi:matrix.org handling event of type RoomHistoryVisibilityEvent
`

**Comment by bpbradley** [comment]
I was able to figure it out. There were changes to how emoji verification works many months ago on the matrix side. The events needed to handle this process don't exist in matrix nio. I was able to get it to work using an unknown event callback and filtering for the correct events. Will try to see if there is a way to patch it into matrix-commander this week if I get some time. 

But yeah --verify will not work in the meantime with an up to date matrix server.
